identity/src/main/java/com/android/identity/GattServer.java

```diff
@@ -321,11 +322,27 @@ public void onCharacteristicWriteRequest(BluetoothDevice device,
                 return;
             }
             mIncomingMessage.write(value, 1, value.length - 1);
+            mLog.transportVerbose(String.format(Locale.US,
+                    "Received chunk with %d bytes (last=%s), incomingMessage.length=%d",
+                    value.length, value[0] == 0x00, mIncomingMessage.toByteArray().length));
             if (value[0] == 0x00) {
                 // Last message.
                 byte[] entireMessage = mIncomingMessage.toByteArray();
                 mIncomingMessage.reset();
                 reportMessageReceived(entireMessage);
+            } else if (value[0] == 0x01) {
+                if (value.length != mNegotiatedMtu - 3) {
```

Comment:
Is it guaranteed that we'll always get this length of data?

That is... even if we negotiated an MTU of 517, does that mean every characteristic read/write should max that? Is it possible that the stack is chunking things up underneath of us?

I don't know BT well enough to understand correct behavior here.

Reply:
The GATT Client can send us any blob they want on any characteristic that they want.

In this case 18013-5 defines the kind of data to send on the Client2Server characteristic. 18013-5 says (8.3.3.1.1.6 Data retrieval):

_If the GATT client wants to send a message to the GATT server, it shall divide the message in parts with a length of 3 bytes less than the MTU size. It then sends these parts to the GATT server using the Write Without Response command via the ‘Client2Server’ characteristic. The first byte of each part is either 0x01, which indicates more messages are coming, or 0x00, to indicate it is the last part of the message._

So, yes, the way I read this is that if it's not the final chunk then it should be size (MTU - 3).

Reply:
Ah, ok. Thanks!

Maybe add a small comment pointing to the section of 18013-5 that mentions the `MTU-3` bit. That was fairtly magical looking to me.

Reply:
I'll abandon this PR and will open a new one which includes this comment and the MTU size change.